mesonbuild/mtest.py

```diff
@@ -1523,9 +1524,52 @@ def __init__(self, options: argparse.Namespace):
             if namebase:
                 self.logfile_base += '-' + namebase.replace(' ', '_')
 
+        self.prepare_build()
+        self.load_metadata()
+
+        ss = set()
+        for t in self.tests:
+            for s in t.suite:
+                ss.add(s)
+        self.suites = list(ss)
+
+    def get_console_logger(self) -> 'ConsoleLogger':
+        assert self.console_logger
+        return self.console_logger
+
+    def prepare_build(self) -> None:
+        if self.options.no_rebuild:
+            return
+
+        if not (Path(self.options.wd) / 'build.ninja').is_file():
+            print('Only ninja backend is supported to rebuild tests before running them.')
+            # Disable, no point in trying to build anything later
+            self.options.no_rebuild = True
+            return
+
+        self.ninja = environment.detect_ninja()
+        if not self.ninja:
+            print("Can't find ninja, can't rebuild test.")
```

Comment:
A better message would be good here. 2 x `can't` is super great in my opinion

Reply:
I agree - I left it that way as I just moved it from one place to another. It's pretty unlikely to be reachable, I guess.  Maybe "Unable to rebuild tests, could not find ninja"?

Reply:
I'll let others weigh in. Didn't realize you just moved it. My bad :)